### ISSUE TITLE
make fails with -Wpedantic

### ISSUE THREAD

**Issue by melissawm** [coin-or/Dip#97]
Hi, 

I'm building Dip on Manjaro Linux with 6.3.1 and GNU Make 4.2.1 and after a successful configure step, I get this error when I run make:

../../../../Dip/src/dippy/DippyDecompAlgo.cpp:32:22: error: ISO C++ forbids converting a string constant to 'char*' [-Wpedantic] 
        pSolutionList);

(there are several more lines like this). I managed to build Dip by commenting the

coin_warn_cflags="-pedantic-errors $coin_warn_cflags"

lines from the configure script, however I don't know if this is supposed to happen.

**Comment by tkralphs** [comment]
What would be most helpful is if you could just create a gist containing the entire output (including the configure step). If you're never done it, just go to https://gist.github.com/ and then paste in the output. You can just copy the link back here.

**Comment by tkralphs** [comment]
The error is this one:
```
error: ISO C++ forbids converting a string constant to ‘char*’. 
```
We are passing string literals to a function that expects `char*`, which is no longer allowed in C++, but is fine in C (which is what Python is written in). Casting the string constants to `char*` or creating a `char*` to pass to the function in the first place should fix it (although these are kind of a hack). E.g., you could try changing
```
PyObject* pResult = PyObject_CallMethod(m_pProb, "chooseBranchSet", "O", pSolutionList);
```
to
```
char * arg1 = "chooseBranchSet", arg2 = "O";
PyObject* pResult = PyObject_CallMethod(m_pProb, arg1, arg2,  pSolutionList);
```
on line 31 of `DippyDecompAlgo.cpp` (and other similar changes to alleviate other errors). You can also switch to `master`, which should work (because -pedantic-errors is disabled). I won't have a chance to play with it for a while, but if you get it working, feel free to submit a pull request.

**Comment by tkralphs** [comment]
This is fixed by 83dd6fc4b56eda8f17e65bf8163324c1ad0c9517 and a number of changes to make Dippy work with Python 3.
